### PR TITLE
4549 browser log viewer

### LINKED ISSUE #4549: 20220915 log file viewer in browser
It would be nice if the log file links across various applications (elegant, srw, flash, etc) would show a modal window with the file contents, with an option to download, rather than the current behavior which is to immediately download the log file.

An example could be found in the flash Problem Files panel which allows viewing the individual files in the problem set. This could be extended to add a "download" icon in the header to allow the user to download the file contents.

Screen Shot 2022-06-22 at 12 04 35 PM

### PR DESCRIPTION
**Pull request by Benbenbenin0**
Fix #4549: Replaced "View log" hyperlink with new directive that views in browser

### CODE REVIEW

File: sirepo/package_data/static/js/elegant.js
Comment (moellep) [inline sirepo/package_data/static/js/elegant.js:1468]: Rather than the nbsp nbsp, you could add a style="margin-left: 1ex" to the close button.

File: sirepo/package_data/static/js/elegant.js
Comment (moellep) [inline sirepo/package_data/static/js/elegant.js:1466]: If you remove the whitespace before the end anchor, the download icon won't show that odd blip when you hover over it.

File: sirepo/package_data/static/js/elegant.js
Comment (moellep) [inline sirepo/package_data/static/js/elegant.js:1508]: You don't need to set anything on appState. You can call setIFrameHTML(data.html) directly.

File: sirepo/template/elegant.py
Comment (robnagler) [inline sirepo/template/elegant.py:1461]: This is a stateful_compute, because it refers to a specific run dir which must have "state" (the log file). 

Here's where it is explained: https://github.com/radiasoft/sirepo/issues/2201#issuecomment-831591025

@e-carlin is there documentation?

File: sirepo/template/elegant.py
Comment (robnagler) [inline sirepo/template/elegant.py:1478]: missing newline at eof. Maybe set up your editor to always put in a newline at eof.

File: sirepo/template/elegant.py
Comment (robnagler) [inline sirepo/template/elegant.py:1470]: text and t temporaries unnecessary. They don't shorten the code, and it's not too deep nesting.

File: sirepo/package_data/static/js/elegant.js
Comment (moellep) [inline sirepo/package_data/static/js/elegant.js:1514]: Don't pass filename to the server - that would allow this API to access any file on the server. Instead, the server side should use the ELEGANT_LOG_FILE constant for the filename.

File: sirepo/package_data/static/html/elegant-visualization.html
Comment (robnagler) [inline sirepo/package_data/static/html/elegant-visualization.html:36]: extra space?

File: sirepo/package_data/static/js/elegant.js
Comment (robnagler) [inline sirepo/package_data/static/js/elegant.js:1452]: unnecessary space? `scope: {}`?

**Review by robnagler — CHANGES_REQUESTED** [review]
minor comments

File: sirepo/template/elegant.py
Comment (moellep) [inline sirepo/template/elegant.py:1478]: You can remove the temporary variables "text" and "t" and just place them inline.

File: sirepo/template/elegant.py
Comment (moellep) [inline sirepo/template/elegant.py:1466]: Use "report=data.model" rather than "animation".

**Review by moellep — COMMENTED** [review]
This is a good start - a few minor things to review.

File: sirepo/template/elegant.py
Comment (e-carlin) [reply sirepo/template/elegant.py:1461]: I think that is an analysis job. statefulCompute is just sirepo-data.json (doesn't go to NERSC for example)

https://github.com/radiasoft/sirepo/wiki/Interacting-with-Simulations-Architecture

File: sirepo/package_data/static/js/elegant.js
Comment (e-carlin) [inline sirepo/package_data/static/js/elegant.js:1495]: I don't believe this check is necessary. I think that appState is guaranteed to be loaded by the time this directive is rendered. Is there a case this is covering that I'm missing?

File: sirepo/package_data/static/js/elegant.js
Comment (e-carlin) [inline sirepo/package_data/static/js/elegant.js:1506]: I think the more "correct" way to get this is to specify `elegantService` in the arguments to the controller function above. Then here to say `elegantService.computeModel()`. 

Every app has a `<app-name>Service` in Sirepo. And every one of those (or almost every one) has an abstraction for getting the computeModel.

File: sirepo/template/elegant.py
Comment (e-carlin) [inline sirepo/template/elegant.py:31]: Sort imports alphabetically.

File: sirepo/template/elegant.py
Comment (e-carlin) [inline sirepo/template/elegant.py:1462]: No need to fix here but I'll fix at some point https://github.com/radiasoft/sirepo/issues/4850

**Review by e-carlin — COMMENTED** [review]
Looks good. Just a few nits.

File: sirepo/package_data/static/html/elegant-visualization.html
Comment (gurhar1133) [inline sirepo/package_data/static/html/elegant-visualization.html:37]: Can remove this comment

**Review by gurhar1133 — APPROVED** [review]
Looks good to me. Good job!

File: sirepo/package_data/static/js/elegant.js
Comment (Benbenbenin0) [reply sirepo/package_data/static/js/elegant.js:1495]: Good point think its unneccesary

File: sirepo/package_data/static/js/elegant.js
Comment (Benbenbenin0) [reply sirepo/package_data/static/js/elegant.js:1506]: Got it good to know.

File: sirepo/package_data/static/js/elegant.js
Comment (e-carlin) [inline sirepo/package_data/static/js/elegant.js:1483]: I don't believe this appState or model check are needed

File: sirepo/package_data/static/js/elegant.js
Comment (e-carlin) [inline sirepo/package_data/static/js/elegant.js:1479]: I think this can be simplified to `$('#sr-text-iframe').attr("srcdoc", html);`. srcdoc is the attribute for setting an html string on an iframe https://www.w3schools.com/tags/att_iframe_srcdoc.asp

File: sirepo/package_data/static/js/elegant.js
Comment (e-carlin) [inline sirepo/package_data/static/js/elegant.js:1498]: Remove debug statement

File: sirepo/package_data/static/js/elegant.js
Comment (e-carlin) [inline sirepo/package_data/static/js/elegant.js:1510]: Since `getDownloadUrl` is only ever called here and there is no other code in this function I think the contents of `getDownloadUrl` can be moved here and that `getDownloadUrl` function can be removed

**Review by e-carlin — APPROVED** [review]
Just a couple nits but looks good. I'll let Paul approve and merge when he reviews.

File: sirepo/package_data/static/js/elegant.js
Comment (moellep) [reply sirepo/package_data/static/js/elegant.js:1483]: appState.isLoaded() will always be true, but model is "appState.models.simulationStatus.animation.computeModel" which gets set with the sim status response, so it may be null initially.

File: sirepo/package_data/static/js/elegant.js
Comment (moellep) [reply sirepo/package_data/static/js/elegant.js:1510]: Also, setIFrameHTML() is a single line method which is called once, so the method could be removed and the code called directly in the sendAnalysisJob data handler.

File: sirepo/package_data/static/js/elegant.js
Comment (moellep) [reply sirepo/package_data/static/js/elegant.js:1498]: You can catch these cases by running "sh test.sh" in the ~/sirepo directory before checking in - it will run jslint on the code, some karma tests, and then the full test suite. For UI changes like this, I usually kill the "sh test.sh" after it gets past the jshint and karma tests because the full test suite takes a long time to run.

**Pull request merged**
